packages/core/src/common/errors.ts

```diff
@@ -20,6 +20,9 @@ export const CONTEXTMENU_WARN_DECORATOR_NO_METHOD =
 
 export const HOTKEYS_HOTKEY_CHILDREN = `${ns} <Hotkeys> only accepts <Hotkey> children.`;
 
+export const ICON_DEPRECATED_NAMES =
+    `${ns} The "pt-icon-" prefix for iconName is deprecated and will be removed in 2.0.`;
```

Comment:
How much tech debt does it allow us to pay down? How many lines of code does it take to maintain compatibility? How much complexity does it add?

IMO, the answer to all those questions is "very little"